js/geo/coordinate.js

```diff
@@ -15,6 +15,7 @@ function Coordinate(column, row, zoom) {
     this.column = column;
     this.row = row;
     this.zoom = zoom;
+    this.posMatrix = null;
```

Comment:
The convenient thing about caching inside `Coordinate` is that coordinate objects are discarded after rendering, so we don't retain outdated calculations in memory. Caching in `calculatePosMatrix` would be much more complicated technically to make it right. I prefer a simple and practical solution rather than ideologically pure one when it's very beneficial.